iothub/device/src/Transport/AmqpIot/AmqpIotTransport.cs

```diff
@@ -177,13 +185,20 @@ private async Task<ClientWebSocket> CreateClientWebSocketAsync(Uri websocketUri,
                     {
                         // Configure proxy server
                         websocket.Options.Proxy = webProxy;
-                        Logging.Info(this, $"{nameof(CreateClientWebSocketAsync)} Setting ClientWebSocket.Options.Proxy");
+                        if (Logging.IsEnabled)
+                            Logging.Info(this, $"{nameof(CreateClientWebSocketAsync)} Setting ClientWebSocket.Options.Proxy");
                     }
                 }
                 catch (PlatformNotSupportedException)
                 {
                     // .NET Core 2.0 doesn't support proxy. Ignore this setting.
-                    Logging.Error(this, $"{nameof(CreateClientWebSocketAsync)} PlatformNotSupportedException thrown as .NET Core 2.0 doesn't support proxy");
+                    if (Logging.IsEnabled)
+                        Logging.Error(this, $"{nameof(CreateClientWebSocketAsync)} PlatformNotSupportedException thrown as .NET Core 2.0 doesn't support proxy");
+                }
+
+                if (_clientOptions?.WebSocketKeepAlive != null)
```

Comment:
AMQP: assign it, if not null.

Reply:
q - does `TimeSpan` default to null or do we need to define it as `TimeSpan?` explicitly?

Reply:
Good point. It must be declared to be nullable.